decidim-bulletin_board-app/app/jobs/sandbox/generate_votes_job.rb

```diff
@@ -4,21 +4,24 @@ module Sandbox
   class GenerateVotesJob < ApplicationJob
     def perform(number_of_votes, election_id, file_path, client_settings = {})
       @election_id = election_id
-      voter_adapter = VotingScheme.from_election(election)[:voter].new(election)
+      @file_path = file_path
+      @client_settings = client_settings
 
+      # Send to the adapter:
+      # - the 'create_election' to provide the details of the election
+      # - the 'end_key_ceremony' to provide the joint key
       %w(create_election end_key_ceremony).each do |message_type|
         log_entry = log_entry_for(message_type)
         voter_adapter.process_message(log_entry.message_identifier, log_entry.decoded_data)
       end
 
-      client = Decidim::BulletinBoard::FileClient.new(file_path, OpenStruct.new(client_settings))
       number_of_votes.times do
         encrypted_ballot = voter_adapter.encrypt(random_plaintext_ballot)
-        client.cast_vote(election_id, SecureRandom.hex, encrypted_ballot)
+        client.cast_vote(election_id, voter_id, encrypted_ballot)
```

Comment:
Maybe we could leave it as it is now, but I think we will have problems with this (now or in the future), as the server should accept several votes from the same voter_id until they are processed.